dali/daliadmin/daadmin.hpp

```diff
@@ -0,0 +1,89 @@
+extern da_decl void count(const char *path);
+
+extern da_decl void dfsfile(const char *lname, IUserDescriptor *userDesc, UnsignedArray *partslist = nullptr);
+extern da_decl void dfspart(const char *lname,IUserDescriptor *userDesc, unsigned partnum);
+extern da_decl void dfsmeta(const char *filename,IUserDescriptor *userDesc, bool includeStorage);
+extern da_decl void setdfspartattr(const char *lname, unsigned partNum, const char *attr, const char *value, IUserDescriptor *userDesc);
+extern da_decl void dfscsv(const char *dali, IUserDescriptor *udesc);
+extern da_decl void dfsCheck();
+extern da_decl void dfsGroup(const char *name, const char *outputFilename);
+extern da_decl int clusterGroup(const char *name, const char *outputFilename);
+extern da_decl void dfsLs(const char *name, const char *options, bool safe = false);
+extern da_decl void dfsmap(const char *lname, IUserDescriptor *user);
+extern da_decl int dfsexists(const char *lname, IUserDescriptor *user);
+extern da_decl void dfsparents(const char *lname, IUserDescriptor *user);
+extern da_decl void dfsunlink(const char *lname, IUserDescriptor *user);
+extern da_decl int dfsverify(const char *name, CDateTime *cutoff, IUserDescriptor *user);
+extern da_decl int dfsperm(const char *obj, IUserDescriptor *user);
+extern da_decl void setprotect(const char *filename, const char *callerid, IUserDescriptor *user);
+extern da_decl void unprotect(const char *filename, const char *callerid, IUserDescriptor *user);
+extern da_decl void listprotect(const char *filename, const char *callerid);
+extern da_decl void checksuperfile(const char *lfn, bool fix);
+extern da_decl void checksubfile(const char *lfn);
+extern da_decl void listexpires(const char * lfnmask, IUserDescriptor *user);
+extern da_decl void listrelationships(const char *primary, const char *secondary);
+extern da_decl void dfscompratio (const char *lname, IUserDescriptor *user);
+extern da_decl void dfsscopes(const char *name, IUserDescriptor *user);
+extern da_decl void cleanscopes(IUserDescriptor *user);
+extern da_decl void normalizeFileNames(IUserDescriptor *user, const char *name);
+extern da_decl void listmatches(const char *path, const char *match, const char *pval);
+extern da_decl void dfsreplication(const char *clusterMask, const char *lfnMask, unsigned redundancy, bool dryRun);
+extern da_decl void migrateFiles(const char *srcGroup, const char *tgtGroup, const char *filemask, const char *_options);
+extern da_decl void getxref(const char *dst);
+
+extern da_decl void listworkunits(const char *test, const char *min, const char *max);
+extern da_decl void workunittimings(const char *wuid);
+extern da_decl void dumpWorkunit(const char *wuid, bool includeProgress);
+extern da_decl void dumpProgress(const char *wuid, const char *graph);
+extern da_decl void wuidCompress(const char *match, const char *type, bool compress);
+extern da_decl void dumpWorkunitAttr(IConstWorkUnit *workunit, const WuScopeFilter &filter);
```

Comment:
check carefully if function are needed externally, or are truly internal utility methods only.
This one isn't for example, it is only called by the other dumpWorkunitAttr.

I think most are called directly from the command line, so do need to be exported, but worth double checking.

Reply:
Sorry, my mistake. Will remove and double check others.